PYIC-8283: Fix device detection to route iPads to desktop flow instead of mobile app download

AmritSidhu opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Fix device detection to route iPads to desktop flow Previously, iPads were incorrectly detected as iPhones due to relying solely on the OS name (iOS) without checking device type. 
This caused users to be routed to the mobile app download page instead of the select device page. 
Updated the sniffPhoneType logic to prioritise device type detection, treating only 'mobile' devices with iOS or Android as phones. 

### Why did it change

All other device types, including 'tablet' (iPads), now correctly fall back to the desktop flow. 
This restores expected behaviour after the recent OS version handling changes.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-8283](https://govukverify.atlassian.net/browse/PYIC-8283)

## Checklists

[PYIC-8283]: https://govukverify.atlassian.net/browse/PYIC-8283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
PYIC-8283: Fix device detection to route iPads to desktop flow instead of mobile app download

Fix device detection to route iPads to desktop flow
Previously, iPads were incorrectly detected as iPhones due to relying solely on the OS name (iOS) without checking device type. This caused users to be routed to the mobile app download page instead of the select device page.
Updated the sniffPhoneType logic to prioritise device type detection, treating only 'mobile' devices with iOS or Android as phones. All other device types, including 'tablet' (iPads), now correctly fall back to the desktop flow.
This restores expected behaviour after the recent OS version handling changes.

thebauSoftwire commented on src/app/shared/deviceSniffingHelper.ts:
```diff
@@ -41,16 +41,22 @@ export const sniffPhoneType = (
+  const osName = parser.getOS().name;
+  const version = parser.getOS().version;
 
-  const version = parser.getOS()["version"];
-  switch (parser.getOS()["name"]) {
-    case OS_TYPES.IOS:
+  // Only treat "mobile" + OS as phone
+  if (deviceType === "mobile") {
+    if (osName === OS_TYPES.IOS) {
```
Do we need to add/update any unit or browser tests for this? E.g. a case for an ipad

thebauSoftwire reviewed (commented):
Nice! Just one comment around tests

AmritSidhu committed:
PYIC-8283: Add test to verify iPad devices are treated as desktop flow in device detection

Added a unit test for sniffPhoneType to ensure iPad user agents do not get incorrectly detected as phones.
This test confirms that iPads return fallback (desktop flow) and prevents regression of the recent fix which prioritises device type over OS for iOS devices.
This ensures iPad users are correctly routed to the select device page rather than the mobile app download flow.
